### PR TITLE
Add go.work support to the documentation

### PR DESCRIPTION
**Pull request by stefanpenner**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Bug fix
> Feature

Documentation
 
> Other

**What does this PR do? Why is it needed?**

Add go.work support to the documentation

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

**Review by fmeum — APPROVED** [review]

File: docs/go/core/bzlmod.md [hunk 1]
```diff
@@ -160,6 +160,24 @@ go_deps.module(
 )
 ```
 
+#### Specifying Workspaces
+
+`go.work` is supported by the go_deps module extension provided by Gazelle.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:165]: Should we mention any restrictions, e.g. the version conflict handling and the fact that go.work is only supported in the root module?

Comment (stefanpenner) [reply docs/go/core/bzlmod.md]: @fmeum I've taken a stab \w a limitations sections, let me know what you think!

File: docs/go/core/bzlmod.md [hunk 2]
```diff
@@ -160,6 +160,28 @@ go_deps.module(
...
+    go_deps,
+    "com_github_gogo_protobuf",
+    "com_github_golang_mock",
+    "com_github_golang_protobuf",
+    "org_golang_x_net",
+)
+```
+
+Limitations:
+* **Root Module Only**: go.work is supported exclusively in the root module.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:182]:
```suggestion
* `go.work` is supported exclusively in the root module.
```

File: docs/go/core/bzlmod.md [hunk 3]
```diff
@@ -160,6 +160,28 @@ go_deps.module(
 )
 ```
 
+#### Specifying Workspaces
+
+The go.work functionality is supported by the go_deps module extension in Gazelle.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:165]:
```suggestion
The `go.work` functionality is supported by the `go_deps` module extension in Gazelle.
```

File: docs/go/core/bzlmod.md [hunk 4]
```diff
@@ -160,6 +160,28 @@ go_deps.module(
...
+    "com_github_gogo_protobuf",
+    "com_github_golang_mock",
+    "com_github_golang_protobuf",
+    "org_golang_x_net",
+)
+```
+
+Limitations:
+* **Root Module Only**: go.work is supported exclusively in the root module.
+* **Cyclical Indirect Dependencies**: Dependencies that are indirect and themselves depend on a go module specified in go.work will have that dependency diverge from the one in go.work. More details can be found here: https://github.com/bazelbuild/bazel-gazelle/issues/1797.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:183]:
```suggestion
* Dependencies that are indirect and depend on a go module specified in `go.work` will have that dependency diverge from the one in `go.work`. More details can be found here: https://github.com/bazelbuild/bazel-gazelle/issues/1797.
```

**Review by fmeum — APPROVED** [review]

**Pull request merged**